prov/efa/src/rdm/rxr_env.h

```diff
@@ -83,10 +85,56 @@ struct rxr_env {
 	int rnr_retry;
 };
 
-extern struct rxr_env rxr_env;
+/**
+ * @brief Get default value of using device's RDMA
+ *
+ * This function gets the default value of using device's RDMA
+ * capability. This function uses the g_device_list initialized
+ * in efa_device_list_initialize(), thus it must be called
+ * after efa_device_list_initialize().
+ *
+ * @return 	1 - Use EFA device's RDMA capability by default
+ * 		0 - Don't use EFA device's RDMA capability by default
+ */
+static inline bool rxr_get_default_use_device_rdma()
+{
+	bool use_device_rdma;
+	uint32_t vendor_part_id = g_device_list[0].ibv_attr.vendor_part_id;
 
-void rxr_env_initialize();
+	/* This is the default scenario. This could be overwritten
+	 * later by the user when fi_param_get_int() is called.
+	 * By default we would not want to use device RDMA in EFA
+	 * gen-0 and gen-1 because it wasn't shipped with RDMA
+	 * capability. Starting from EFA gen-2, if the device
+	 * supports RDMA, use it. */
+	if (vendor_part_id == 0xefa0 || vendor_part_id == 0xefa1) {
+		use_device_rdma = false;
+	} else {
+		use_device_rdma = true;
+		if (!(g_device_list[0].efa_attr.device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
+			EFA_WARN(FI_LOG_FABRIC,
+				"EFA device with vendor_part_id 0x%x has no rdma-read capability.\n", vendor_part_id);
+			use_device_rdma = 0;
+		}
+	}
+
+	return use_device_rdma;
+}
 
 int rxr_env_get_use_device_rdma();
 
+static inline bool rxr_env_or_default_use_device_rdma() {
```

Comment:
I think we need `get` in the name

Reply:
the name should match the file too right?  so: `rxr_env_get_default_or_env_use_device_rdma()` ?

Reply:
It seems we can merge this one with `rxr_env_get_use_device_rdma()` and have a standalone `rxr_env_has_use_device_rdma()`?

Reply:
Good idea.  Done.